sdk/core/abort-controller/src/AbortSignal.ts

```diff
@@ -35,6 +31,13 @@ export interface AbortSignalLike {
   ): void;
 }
 
+type AbortEventListener = (this: AbortSignalLike, ev?: any) => any;
+
+// eslint-disable-next-line no-use-before-define
+const listenersMap = new WeakMap<AbortSignal, AbortEventListener[]>();
```

Comment:
I'm confused how this gets used before it's defined? it looks like the only thing above it is the definition of AbortSignalLike

Reply:
Yeah this rule applies now on types as well, see the list of linting changes in this PR description. Perhaps we should configure it not to? here is the docs: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-use-before-define.md

Reply:
never mind, I used the typescript-eslint's rule instead and it is smart enough to not report an error here.